File: 349.cpp
#include<iostream>
#include<algorithm>
using namespace std;

class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        vector<int> res;
	set<int> s(nums2.begin(),nums2.end());
        nums2.assign(s.begin(),s.end()); 
        for(int num:nums1){  
            vector<int>::iterator position=std::find(nums2.begin(),nums2.end(),num);
            if(position!=nums2.end()){
                res.push_back(*position);
                nums2.erase(position);
            }   
        }
        return res;
    }
};

File: 349.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        res=[]
	for num in nums1:
	    if num in nums2:
                res.append(num)
                nums2=filter(lambda a:a!=num,nums2)
        return res

s=Solution()
print(s.intersect([1,2,2,1],[2,2]))
